### ISSUE TITLE
Inconsistent type inference with `type(None)`

### ISSUE THREAD

**Issue by smackesey** [python/mypy#12658]
**Bug Report**

```
TYPE_NONE = type(None)
reveal_type(type(None))  # => `Type[None]`
reveal_type(TYPE_NONE)  # => `builtins.object`
```

**Expected Behavior**

The two `reveal_type` calls above should print the same thing (`Type[None]`)

**Your Environment**

- Mypy version used: 0.942
- Python version used: 3.10.4
- Operating system and version: macOS 12.3.1

**Comment by Zeckie** [comment]
Is that really and different from this?

```python
one = 1
reveal_type(1)  # note: Revealed type is 'Literal[1]?'
reveal_type(one)  # note: Revealed type is 'builtins.int'
```

**Comment by smackesey** [comment]
Not sure-- probably share a similar root but IMO the example I posted is even less intuitive (one is a `Type`, the other is `builtins.object`) because this pattern does not occur when you use a different type (like `int` in place of `type(None)`).

**Comment by brianschubert** [comment]
As of #16315 (released with mypy v1.9.0), mypy infers `type[None]` for both cases.
